asabiris/formatter/jinja/service.py

```diff
@@ -37,21 +36,22 @@ def create_nested_dict_from_dots_in_keys(data):
 
 	:param data: The input dictionary that may contain keys with dots in them, indicating nested levels
 	of dictionaries
-	:return: a nested dictionary where keys containing dots (".") have been split into sub-dictionaries.
+	:return: A nested dictionary where keys containing dots (".") have been split into sub-dictionaries.
 	"""
-	nested_dict = {}
-	for key, value in data.items():
-		if isinstance(value, dict):
-			value = create_nested_dict_from_dots_in_keys(value)
-
-		if '.' in key:
-			parts = key.split('.')
-			current_dict = nested_dict
-			for part in parts[:-1]:
-				current_dict.setdefault(part, {})
-				current_dict = current_dict[part]
+	nested_dict = {}  # Initialize an empty nested dictionary
+	stack = [(nested_dict, key, value) for key, value in data.items()]  # Create a stack of (dictionary, key, value) tuples
 
-			current_dict[parts[-1]] = value
+	while stack:
```

Comment:
Cool. I should read more about this structure. However, the recursion is missing here, I am afraid. This dictionary is not processed right: 
```
{'alert': {'event.other.id': '1234'}}
```

Reply:
@eliska-n I will test it more! Thanks Eliska

Reply:
@eliska-n I will see what I can do.